docassemble/ALToolbox/data/questions/display_template_demo.yml

```diff
@@ -1,42 +1,97 @@
+  1. **copy** (default=False) 
+      - It adds a copy button under the template for users to copy/share the template content.
+      - This feature applies the **copy_button_html()** function to template content. See [copy button demo](${interview_url(i=user_info().package + ":copy_button_demo.yml")}).
+  1. **class_name** (default=None) 
+      - To style your template, include a class_name input in the call, and add your css rules either on the screen or to a .css file.  
+      
+  You can mix and match these features. The following screens will show you how. 
   
-  1. If your markdown template content is long, you can use this tool to display it with a stroll bar on the screen. 
-  1. If your markown template has a subject line in addition to content, this tool will display both with one line of code rather than two. 
+  &#128276 To implement this function, add **display_template.yml** in the _include_ section of your interview and follow the examples in this demo.
+---
+id: scrollable only examples
+continue button field: scrollable_only_examples
+question: |  
+  Scrollable examples
+subquestion: |    
+  ##### Srollable only
+  ${ display_template(my_template1) }<br>
   
-  ####Notes:
+  ##### Style only
+  ${ display_template(my_template2, scrollable=False, class_name="my_color")}
+---
+id: collapse and scrollable examples
+continue button field: collapse_n_scrollable_examples
+question: |  
+  Collase and scrollable examples
+subquestion: |
+  ##### Collapse only with a style
+  ${ display_template(my_template1, scrollable=False, collapse=True, class_name="my_color") }
   
-  1. Template's "subject" line will be displayed automatically, but it's NOT required to use this tool. Your template can have just "content".
-  1. You can turn off the scrollable feature for short content.
-  1. To style the panel, add a class name in the call, then add your css rules either on the screen (like in this demo) or add them to a .css file.
-    
-  ${ display_template(my_template, scrollable=True, class_name="my_color") }<br>
-    
-  ${ display_template(my_template, scrollable=False) }  
-css: |
-  <style>
-    .my_color {
-      background: lightcyan; 
-      color: darkgreen;      
-    }
-  </style>
-
+  ##### Collapse and scrollable, without the copy button
+  ${ display_template(my_template2, collapse=True) }
 ---
-template: my_template
+id: copy button examples
+continue button field: copy_button_examples
+question: |
+  Copy button examples
+fields:  
+  - html: <h5>Copy and scrollable</h5>
+  - note: |      
+      ${ display_template(my_template1, copy=True) }
+  - html: <h5>Copy and collapse but not scrollable</h5> 
+  - note: | 
+      _(Hover over the box to see the full template content. If this is annoying, keep the default value scrollable=True as shown in the very last example.)_<br><br>
+      ${ display_template(my_template1, scrollable=False, collapse=True, copy=True) }  
```

Comment:
I played around with things, trying to get the box to have the correct height to show all of the content without requiring a mouse over, and I think the issue is that `my_template1.content_as_html(...)` returns everything in a textarea (I can't find where that happens in docassemble), and it seems that textareas in Firefox aren't resizable with CSS. Changing the textarea to a div, and just using `height: auto` in the CSS for that element works great, without any sort of sudden changes. 

I couldn't find a better solution, just wanted to document what I had found.

Reply:
I know what you mean. I tried <div> as well, it would work for the height, but won't trigger the button click reaction, because <div> is not an input type of field. (sigh)

Reply:
If we want to copy from a `div` instead of a `textarea`, it may be possible as long as we do it a different way: https://stackoverflow.com/a/48020189/14144258.  We can also, as you folks have said, make it an issue for the future and I may have time to try it out then.

Reply:
I'm looking at the js code and I think I'm not understanding what the problem is. I thought it was something wrong with the copy to text, but there's something about the copy button itself? Maybe we can talk about it during deep dive today.

Reply:
It's both: copy and tooltip would both stop working.

Reply:
I'll resolve this conversation and create a new issue for changing `textarea `to `div`